### ISSUE TITLE
The new economy: some observations

### ISSUE THREAD

**Issue by Vectorial1024** [DeadAirRT/deadair_eco#3]
I got some hours with this mod enabled, and I am observing a new economic pattern in the game.

---

If I am not mistaken, the idea of this mod (compared with previous versions of "economy mod" also by you, of course) is that, instead of having large NPC megastations to produce many resources, there are new secondary resources so that roughly the same amount of wares are produced per hour by NPC.

This is good to maintain economic stability because shipyards are now less likely to starve, but this also has the side effect of general overproduction of wares in the entire galaxy. The production efficiency is so high, it is not profitable to mine minerals, and it is also not exactly profitable to expand production since basic wares are not being consumed fast enough.

This also indirectly caused the discovery of the minprice bug since everything is being sold at absolute minimum price (except perhaps hull parts; it manages to stay somewhat at near 0% selling price).

---

A separate observation is that, for some reason, only just a few hours into the game, several NPC trade stations already have their Nividium storage at 75% full, and so Nividium is only bought at min price. Can't use Nividium mining to transition to mid-game.

---

Are these something you have intended? 

The early game has become easier (e.g. extreme trade margins for some wares), but it seems less obvious on how to transition to mid-game other than perhaps building a hull parts production or starting some recycling business in the Avarice system.

**Comment by DeadAirRT** [comment]
The goal of the new wares and workforce changes are to increase importance of workforce and to reduce the number of stations required for a fully functioning economy. Looking at Eco on it's own, since I no longer balance the economic production at game start there is imbalance because there has always been imbalance. When you look at eco + scripts DA God, the numbers set by default that factions build up to are the observed ware usage (a lot of data gathering) for a shipyard+wharf and the amount of wares to complete the chain. There is a bit of overproduction even on my final balanced economy due to transportation bottlenecks and of course, war interruptions.

There were several concessions I had to make for the sake of the user. I maintained station calculator compatibility sans workforce. I also had to consider performance, as more economy ships and stations means more cpu stressing.

Depending on what you consider the mid-game transition, the mod enables building the new ware production stations which require minimal capital investment, nividium mining is still viable using the DA Fill drain (which empowers the faction as well instead of just existing), increased trading capacity due to economic throughput, and of course if DA God is activated there are quite a few opportunities for ware supply that shifts over time.

This mod is definitely not an end-all be-all by itself, as is the case with any xml diff mod.

**Comment by DeadAirRT** [comment]
So I'm in the process of testing my adjustment to pricing calculation and it seems to revert the time scaling changes that caused prices to be mostly at extremes. You can add this to parameters.xml in Eco if you would like to check it out:
<replace sel="/parameters/economy/prices/product/@factor">50000000</replace>

**Comment by Vectorial1024** [comment]
That's quite unexpected. I have always thought trade ware prices are calculated "internally" and is largely unchangeable or unexplainable.

**Comment by Malorn-Deslor** [comment]
I would add my voice that the economy seems very...closed. I don't see the AI requiring much in the way of input in normal circumstances. Miners especially are hard hit, often having nothing to do and nowhere to sell early game. I had one auto-miner who couldn't sell anything for nearly 6 hours after the game began, and he was staged in Argon space with a range of 5 jumps. In the mid-game I am seeing some buy offers, but still pretty low price and for amounts in the low hundreds. The AI miners are clearly working very effectively, I'd wager.

I am not using DAgod, so station balance may differ. But it does seem like it's hard to 'break-in' to trading from the outside, likely because stations are larger and thus shuffle goods less often, which is a good thing. But goods are heavily in surplus in the early game, which is just when the player wants to trade with AI stations. Looking at the DAfill logs, there are huge amounts of some resources being removed, especially all types of food and medical supplies.

Notably, the things being added in large amounts were hull parts, engine parts, and weapon/turret components. These are all ship production uses, and probably means that the production isn't keeping up and other resources are thus stacking up. Claytronics, however, are being removed at a huge rate, meaning that the AI COULD be constructing stations, but isn't.

I don't have DAgod enabled, because I couldn't quite understand what it was doing and it said it was not reversible if enabled. Perhaps that's the problem and the AI isn't rebuilding the right stations properly? What exactly does DAgod do, does it just set up production module quotas as it appears to do? Or is there something more sneaky going on?

**Comment by Vectorial1024** [comment]
DA God is irreversible only because the god file controls the initial conditions of the game (how many stations) and after the game is set, stations must be constructed/destroyed by real in-game mechanics.

---

Some additional observations now that I got my game into mid-game:

I got my first bucket of gold via supplying Refined Metals to a Hull Parts factory. If it weren't for this, I would have struggled to get anything up and running because as we have noticed, early game has so much surplus there aren't really any good trade opportunity.

The reduced number of spawned production modules has indirectly nerfed black marketeers because, it turns out, black market drug demand is linked with station workforce amount. ref: https://forum.egosoft.com/viewtopic.php?p=5239488

This extreme surplus of wares is also indirectly killing the VIG/RIP economy. I don't know why, or which mod/feature is controlling this, but apparently:
- Stations which are stuck with no production for too long may be scrapped
- Claytronics are generally unneeded early game, thus blocking Hull Parts production (!!) and increasing its price
- Since Claytronics are in extreme surplus, Scrap Processor stations may randomly be scrapped
- Now there is an unrecoverable Hull Parts shortage

But this extreme surplus has its good side: PHQ research is now quite cheap to do, which helps with unlocking ship/weapon mods and teleportation.

**Comment by Malorn-Deslor** [comment]
Thanks for the info on the god file, I should probably activate that so the AI starts building properly.

Overall I still think things are improved, since I am seeing a lot more movement in the AI economy, even if I can't be part of it easily.

**Comment by DeadAirRT** [comment]
I'll try to look into seeing if there is anything I can do to alleviate the economy issues when not used in conjunction with DA Scripts God. One of the main intentions behind the changes are to make it so less stations are required overall for a fully functional economy (which is not how I would describe vanilla god.xml quotas). Raw materials are definitely a more challenging opportunity than desired, but it's pretty difficult to balance for all the variables.

@Vectorial1024 - I'll definitely look into adjusting the factionlogic for station deactivation/removal ASAP. The scripts are quite complex so it may take a bit of time. Right now I believe the only reaction that vanilla logic handles is:
Overproduction == Pause station -> if another station within set range after period of time -> recycle station
Shortage == commandeer freighter -> if shortage isn't solved -> add one module -> if shortage isn't solved -> build new station with 1-2 modules

@Malorn-Deslor - the DA Scripts God portion can seem a bit daunting at first, but it basically is just new logic I wrote that makes factions try to build the number of production modules in the settings, and if they are lost, replace them. If you need any help or any parts of the menu aren't clear, please let me know.

**Comment by Vectorial1024** [comment]
Oh, so it is the vanilla faction logic that controls what stations to scrap.

Then, @DeadAirRT perhaps we should delay touching it and instead ask about this on EgoSoft forums; I think they will be interested in fixing the VIG/RIP imbalance.

Tbf when I was playing the DLC sth like a year ago, I thought the station destructions inside Avarice are lore accurate because they couldn't afford to pay for Protectyon, so I didn't report anything to anyone. Now I know.

**Comment by Vectorial1024** [comment]
Still, I have some ideas on what might be done to improve the situation (assuming God mod is also used), and I'm just going to open new issues describing specific suggestions.

**Comment by Malorn-Deslor** [comment]
> @Malorn-Deslor - the DA Scripts God portion can seem a bit daunting at first, but it basically is just new logic I wrote that makes factions try to build the number of production modules in the settings, and if they are lost, replace them. If you need any help or any parts of the menu aren't clear, please let me know.

So I since activated it, been playing with it for about 6 hours now. I see a huge number of new stations popping up everywhere. Honestly, I don't know how good of a testbed my game is, since the Xenon are rampaging and destroying stations all over even in the depths of argon space. They may be trying to keep up with replacements and never managing to finish them, it's hard to say. But the economy seems more varied and I am seeing desired goods in a lot more places now. Likely because freighters are dying constantly to xenon fighter swarms.

None of that is bad, btw, just the nature of the game playing out. I don't even have xenon parts of the scripts active, so they are doing this on their own dime.

The only concern I would have with DAgod is what happens when they go PAST the limits in the file? For example, if the terrans expanded heavily and doubled in territory, they wouldn't add more stations as I understand it? And if a nation was knocked back to a single sector, they would try to build all their stations in that single sector? Which might be good for them, honestly, but they would be unlikely to find the raw materials to fuel themselves,

Of course they never build more wharfs or shipyards either, so perhaps that wouldn't matter since there is only so many ship building slots possible. It's a problem that may not be resolvable via modding, and the factions never really can grow past certain limits. In the base game not a huge problem, but with the war scripts I have seen some real border changes occur.

The only way I could see being able to fix it is somehow scale the modules the nation intends to build to the number of sectors they control, but that may simply be impossible to do with modding. A challenging question for a game that really isn't meant to change in that way.

**Comment by DeadAirRT** [comment]
@Malorn-Deslor So in the future, it may be possible to expand the logic to be based on how large the faction grows but my usual method is to start with something functional and come up with ways to expand it over time. It has quite aggressive spreading for where new stations should be built for factions that have multiple sectors so I'm glad to hear that it is noticeable. The script itself is quite complex due to the amount of variables and data scrubbing that has to occur (especially since I let player adjust the amounts) while also balancing performance impact of the script and even more importantly the impact of additional stations.

The vanilla logic will basically never actually expand much due to it's current feature limitations. Right now, vanilla logic will only perform the logic I mentioned briefly in the portion directed at vectorial. I feel at the moment my script does a good job of helping factions build up to a level where there is actually resources available for ship building and makes the universe more volatile.

The factions are capable of expanding shipyard L production capacity with an option in the god options menu but it's definitely a feature I may look at in the future. I did have an error in my formula to calculate how many modules of each type were necessary for a balanced economy with a bit of overproduction (including shipyard and wharf resource usage which has to be captured via a script and monitored for quite a long time to get useful trends/data) so they go a bit overboard if there is a lack of conflict/interruption. Vanilla module counts (running just eco but not using the DA God expansion), still has factions only supplying a portion of what one shipyard+wharf require to be productive constantly.

**Comment by Malorn-Deslor** [comment]
Thank you for the insight. I've modded before, so I understand that there are countless limitations to what is possible, and the extent of your changes already are profound in nature. I did not mean to imply that what existed was not a significant improvement.

Out of curiosity, can it write new values mid-game? So for example if I saw a massive expansion would it be possible to increase that factions intended number of modules manually?
